include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -3423,7 +3380,28 @@ inline TriaIterator<TriaAccessor<dim - 1, dim, spacedim>>
 CellAccessor<dim, spacedim>::face(const unsigned int i) const
 {
   AssertIndexRange(i, this->n_faces());
-  return dealii::internal::CellAccessorImplementation::get_face(*this, i);
+  if constexpr (dim == 1)
+    {
+      TriaAccessor<0, 1, spacedim> a(
+        &this->get_triangulation(),
+        ((i == 0) && at_boundary(0) ?
+           TriaAccessor<0, 1, spacedim>::left_vertex :
+           ((i == 1) && at_boundary(1) ?
+              TriaAccessor<0, 1, spacedim>::right_vertex :
+              TriaAccessor<0, 1, spacedim>::interior_vertex)),
+        internal::TriaAccessorImplementation::Implementation::vertex_index(
+          *this, i));
```

Comment:
I wanted to get rid of this call too (i.e., use `this->vertex_index()`) but that causes some tests to fail - I'll handle that set of overloads in a future PR.